k9mail/src/main/java/com/fsck/k9/activity/compose/RecipientPresenter.java

```diff
@@ -889,14 +938,12 @@ private boolean canHandshakeSingleAddress(
             List<Address> newCcAdresses,
             List<Address> newBccAdresses
     ) {
-        if (!newBccAdresses.isEmpty()) return false;
-        List<Address> candidates = new ArrayList<>();
-        candidates.addAll(newToAdresses);
-        candidates.addAll(newCcAdresses);
-        return candidates.size() == ONE_ADDRESS
+        return newToAdresses.size() == ONE_ADDRESS
+                && newBccAdresses.isEmpty()
+                && newCcAdresses.isEmpty()
                 && PlanckUtils.isRatingReliable(privacyState)
                 && account!= null && account.isPlanckPrivacyProtected()
-                && !candidates.get(0).getAddress().equalsIgnoreCase(account.getEmail()); // recipient not my own account
+                && !newToAdresses.get(0).getAddress().equalsIgnoreCase(account.getEmail()); // recipient not my own account
```

Comment:
funnily in Outlook you can reset your own and it triggers a key reset, but I prefer to hide the option like we do here